src/client/src/commons/form/profile/components/profileLayers/components/chronoStratigraphyLayers/lithologyLayers.js

```diff
@@ -0,0 +1,236 @@
+import React, { useEffect, useRef } from "react";
+import * as Styled from "./lithologyLayersstyles";
+import PropTypes from "prop-types";
+import Draggable from "react-draggable";
+import { NumericFormat } from "react-number-format";
+import { useLayers } from "../../../../../../../api/fetchApiV2";
+import { CircularProgress } from "@mui/material";
+
+const LithologyLayers = props => {
+  const { stratigraphyId, navigationState, setNavigationState, unit } = props;
+
+  const { data } = useLayers(stratigraphyId);
+
+  const element = useRef(null);
+
+  useEffect(() => {
+    updateDimensions();
+    window.addEventListener("resize", updateDimensions, { passive: true });
+    return () => {
+      window.removeEventListener("resize", updateDimensions);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const handleColor = layer =>
+    `rgb(${(
+      JSON.parse(layer?.lithostratigraphy?.conf ?? null)?.color ?? [
+        255, 255, 255,
+      ]
+    ).join()})`;
+
+  const handlePattern = layer =>
+    `url(/img/lit/${JSON.parse(layer?.lithology?.conf ?? null)?.image})`;
+
+  const handleOnWheel = event => {
+    event.stopPropagation();
+
+    let scale = navigationState?.scale;
+    let factor = 0.05;
+    if (scale >= 0.75) {
+      factor = 0.1;
+    } else if (scale >= 0.5) {
+      factor = 0.05;
+    } else if (scale >= 0.2) {
+      factor = 0.025;
+    } else if (scale >= 0.05) {
+      factor = 0.01;
+    } else if (scale < 0.05) {
+      factor = 0.005;
+    }
+
+    if (event.deltaY < 0) {
+      scale -= factor;
+    } else {
+      scale += factor;
+    }
+
+    const rangeHeight = scale * navigationState?.height;
+
+    let topOfLense = navigationState?.top;
+    if (navigationState?.top + rangeHeight > navigationState?.height) {
+      topOfLense =
+        navigationState?.top -
+        (navigationState?.top + rangeHeight - navigationState?.height);
+    }
+
+    setNavigationState(prevState => ({
+      ...prevState,
+      scale: Math.min(Math.max(0.02, scale), 1),
+      top: topOfLense < 0 ? 0 : topOfLense,
+    }));
```

Comment:
PP: Kommentar hinzufügen, der kurz schildert, was hier abgeht.

Reply:
Das habe ich selbst nicht ganz verstanden. Da es in etwa das richtige macht habe ich es nicht neu geschrieben.

Reply:
Alles klar. Wusste nicht, dass der Code übernommen wurde.